Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py

```diff
@@ -868,27 +875,53 @@ def fetch_incidents(client, last_run, first_fetch_time, min_severity):
+    demisto.debug(f"last fetch time is: {last_fetch_time}, last fetch ids are:{last_fetch_ids}")
 
     # Handle first time fetch
-    if last_fetch_time is None:
+    if last_fetch_time is None:  # this is the first fetch, or the First fetch timestamp was reset
+        demisto.debug("im in last fetch time is none")
+        # handle via time stamp
         last_fetch_time_str, _ = parse_date_range(first_fetch_time, DATE_FORMAT)
         last_fetch_time = dateparser.parse(last_fetch_time_str)
+        latest_created_time = last_fetch_time
+        latest_created_time_str = latest_created_time.strftime(DATE_FORMAT)
+        command_args = {'filter': f'properties/createdTimeUtc ge {latest_created_time_str}'}
+        first_fetch = True
+
     else:
+        demisto.debug("handle via id")
+        # handle via id
         last_fetch_time = dateparser.parse(last_fetch_time)
+        latest_created_time = last_fetch_time
+        command_args = {'filter': f'properties/incidentNumber ge {last_incident_number}'}
+        first_fetch = False
 
-    latest_created_time = last_fetch_time
-    latest_created_time_str = latest_created_time.strftime(DATE_FORMAT)
-    command_args = {'filter': f'properties/createdTimeUtc ge {latest_created_time_str}'}
-    command_result = list_incidents_command(client, command_args, is_fetch_incidents=True)
-    items = command_result.outputs
+    demisto.debug(f"command_args:{command_args}, first_fetch:{first_fetch}")
+    command_result = list_incidents_command(client, command_args, is_fetch_incidents=True, first_fetch=first_fetch)
+    raw_incidents = command_result.outputs
+    demisto.debug(f"these are the raw incidents:{raw_incidents}")
+
+    next_run, incidents = process_incidents(raw_incidents, last_fetch_ids, min_severity, latest_created_time, last_incident_number)
+    return next_run, incidents
+
+
+def process_incidents(raw_incidents, last_fetch_ids, min_severity, latest_created_time, last_incident_number):
+    demisto.debug(f"starting to process incidents")
+    demisto.debug(f"latest_created_time: {latest_created_time}, last_incident_number:{last_incident_number}")
     incidents = []
     current_fetch_ids = []
+    if not last_incident_number:
+        last_incident_number = 0
 
-    for incident in items:
+    for incident in raw_incidents:
         incident_severity = severity_to_level(incident.get('Severity'))
+        demisto.debug(f"incident id: {incident.get('ID')}, severity level:{incident_severity}, incident number:{incident.get('IncidentNumber')}")
 
         # fetch only incidents that weren't fetched in the last run and their severity is at least min_severity
-        if incident.get('ID') not in last_fetch_ids and incident_severity >= min_severity:
+        # and their id is larger then the previous incident
+        if incident.get('ID') not in last_fetch_ids and incident_severity >= min_severity and last_incident_number != incident.get('IncidentNumber'):
```

Comment:
I think we might want to break it down into two functions -> by id order and by time order. Here we have two different flows combined, and it isn't apparent.